src/odemis/util/angleres.py

```diff
@@ -427,3 +435,166 @@ def _CreateMirrorMask(data, pixel_size, pole_pos, offset_radius=0, hole=True):
+    """
+    Calculates the polar representation (angle: phi, radius: theta) from the rectangular
+    representation (theta, phi) of the raw data.
+    :param data: (DataArray) The data array containing the image. Shape is (theta, phi).
+    :param colormap: (int) Number of the colormap. TODO this is not very generic yet...
+    :return: (DataArray) Shape is (x, y).
+    """
+    # TODO size and dpi should be calculated based on input to match the 800?
+    output_size = 1134
+    sizefig = (10, 10)
+
+    # For each pixel of the input ndarray, calculate the corresponding theta, phi values.
+    phi_indices = numpy.linspace(0, 2 * numpy.pi, numpy.size(data, 1))  # list of px indices (1D array)
+    theta_indices = numpy.linspace(0, numpy.pi / 2, numpy.size(data, 0))  # list of px indices (1D array)
+    phi, theta = numpy.meshgrid(phi_indices, theta_indices)
+    y_data_polar = numpy.cos(phi) * theta
+    x_data_polar = numpy.sin(phi) * theta
+
+    # Calculate the mirror mask in rectangular phi/theta representation
+    # TODO hole is now always False...change to True?
+    mask = _CreateMirrorMask_Rectangular(data, theta_indices, phi_indices, hole=False)
+    # Get the upper/lower intensity limit, by checking for a fixed amount of outliers.
+    lim1, lim2 = img.getOutliers(data[mask == 1], outliers=0.0039)  # magical number DONOT change!! 1/256
+    data_masked = data.copy()
+    # Mask data (everything outside of mask is set to nan) to reduce calculation time
+    data_masked[mask == 0] = numpy.nan
+
+    # plot the data
+    fig = plt.figure(figsize=sizefig, dpi=80)
+    if colormap == 1:
+        colormap = mitcolors()
+    elif colormap == 2:
+        colormap = EELSmap()
+    plt.pcolormesh(x_data_polar, y_data_polar, data_masked, cmap=colormap, vmin=lim1, vmax=lim2)
+    plt.axis("square")
+
+    # get the data from the figure canvas as numpy array
+    result = _figure2data(fig)
+    plt.close(fig)  # to prevent memory leakage
+
+    data.metadata.update({model.MD_DIMS: 'YXC'})
+
+    result = model.DataArray(result, data.metadata)
+
+    return result
+
+
+def _figure2data(figure):
+    """
+    Extracts the data from the figure canvas and stores it in an numpy array.
+    ":param figure: (matplotlib figure) Figure to extract the data from.
+    :return: (ndarray) Array containing the image from the plotted figure.
+    """
+    # TODO how to fit image with overlay?? Looks like slightly shifted to bottom..
+
+    figure.set_facecolor((0, 0, 0))
+    figure.gca().axis("tight")
+    figure.gca().set_xticks([])
+    figure.gca().set_yticks([])
+    figure.gca().axis('off')
+    figure.tight_layout(pad=0)
+    figure.canvas.draw()
+
+    w, h = figure.canvas.get_width_height()
+    image = numpy.fromstring(figure.canvas.tostring_argb(), dtype=numpy.uint8)
+    image = image.reshape(h, w, 4)
+    image = image[:, :, 1:]
+    return image
+
+
+# TODO do we want to have the colormaps in a separate special file for odemis?
```

Comment:
Indeed, if we support colormaps for any greyscale projections, they should go to a separate file. Let's leave them here for now.